### PR TITLE
Fixes #18243: Add solaris package support to ncf

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/18243

### CODE REVIEW

**Comment by peckpeck** [comment]
PR rebased

**Comment by peckpeck** [comment]
Commit modified

File: tree/30_generic_methods/package_state.cf [hunk 1]
```diff
@@ -125,9 +125,9 @@
 # @parameter_constraint version "allow_empty_string" : true
 # @parameter architecture Architecture of the package, can be an architecture name  or "default" (defaults to "default")
 # @parameter_constraint architecture "allow_empty_string" : true
-# @parameter provider Package provider to use, can be "yum", "apt", "zypper", "zypper_pattern", "slackpkg", "pkg" or "default" for system default package manager (defaults to "default")
+# @parameter provider Package provider to use, can be "yum", "apt", "zypper", "zypper_pattern", "slackpkg", "pkg", "ips" or "default" for system default package manager (defaults to "default")
```
Comment (amousset) [inline tree/30_generic_methods/package_state.cf:128]: Could you add in the docs that local package install does not work?

**Comment by peckpeck** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

**Pull request merged**